### PR TITLE
using the cmake bundled in visual studio

### PR DESCRIPTION
**Pull request by wenbingl**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -22,9 +22,10 @@
 
 def load_msvcvar():
     if os.environ.get('vcvars'):
+        os.environ['__vcvars'] = '"{}"'.format(os.environ['vcvars'])
         stdout, _ = subprocess.Popen([
-            'cmd', '/q', '/c', '(%vcvars% & set)'],
-            stdout=subprocess.PIPE, shell=True, universal_newlines=True).communicate()
+            'cmd', '/q', '/c', 'call %__vcvars% && set'],
+            stdout=subprocess.PIPE, shell=False, universal_newlines=True).communicate()
```
Comment (skottmckay) [inline setup.py:28]: Does it need to create a new environment variable, or would just setting `shell=False` be enough?

File: .az/mshost.yaml [hunk 2]
```diff
@@ -317,6 +317,7 @@ jobs:
 
       - script: |
           call activate pyenv
+          $env:vcvars="$env:VCINSTALLDIR\Auxiliary\Build\vcvars64.bat"
```
Comment (skottmckay) [inline .az/mshost.yaml:320]: Might be good to add a comment saying we do this so cmake comes from VS

**Review by skottmckay — APPROVED** [review]

File: setup.py (hunk 1 above)
Comment (wenbingl) [reply setup.py:28]: I think shell=True is to create another cmd to execute the command line 'cmd /q /c ...', which looks redundant.

File: .az/mshost.yaml (hunk 2 above)
Comment (wenbingl) [reply .az/mshost.yaml]: yes. I will put some comments here after the tryout of getting where VS is installed

File: .az/mshost.yaml [hunk 3]
```diff
@@ -317,7 +317,16 @@ jobs:
 
       - script: |
           call activate pyenv
-          set vcvars=%VCINSTALLDIR%Auxiliary\Build\vcvars64.bat
+          @echo find where Visual Studio was installed and use the bundled cmake
```
Comment (snnn) [inline .az/mshost.yaml:320]: You can try this code : https://github.com/microsoft/onnxruntime-inference-examples/blob/main/.pipelines/OneBranch.Official.yml#L114

**Comment by wenbingl** [comment]
@skottmckay , @snnn who can help with an approval?

**Review by xadupre — APPROVED** [review]

**Comment by wenbingl** [comment]
@xadupre , Thanks!

**Pull request merged**